### ISSUE TITLE
Configuration parser returns empty list

### ISSUE THREAD

**Issue by alecmev** [emersion/kanshi#15]
The config:

```ini
{
	output eDP-1 scale 1.0
}
```

Printed here:

https://github.com/emersion/kanshi/blob/ed21acce0c52f7893c903f46b09b4a3b55e2c198/src/main.rs#L161

The output:

```
Connected outputs:
output eDP-1 ...
output DP-4 ...
output DP-3 ...
Saved configurations: []
...
```

Tested on both master and [the last commit of 2017](https://github.com/emersion/kanshi/tree/dc905ae76892070cc72f6dc5142cd88041c4f27d). Latest everything.

I've gone through `store.rs` a couple of times, but was unable to find any gaping holes. I assume you're using this util yourself, and it's working fine for you? Configured through `KanshiStore`?

**Comment by alecmev** [comment]
@emersion What a coincidence that fractional scaling wasn't supported... :smile: But this wasn't the issue I was reporting, this was just a random small repro config.

I've done some extra testing, and, for some reason, now I'm not able to get `kanshi` to run without panicking at all, whatever config I use ("now" = time-wise, not last commit's fault). The error it panics with is `Incomplete(Size(1))`, on any config, even on the one from the post above (wouldn't panic on it yesterday). I've done some googling, and, it appears, it could be related to `nom` 3 vs. `nom` 4, as mentioned in [this issue](https://github.com/Geal/nom/issues/751). So I have changed its version back to `3.2.1`, and now everything is working fine.

About the panics, since `nom` 4 changes the behavior around partial parsing, and it could be that the buffer is processed in chunks for some reason (maybe even somehow related to the environment... having a hard time imagining how), it just receives an incomplete file and fails.

Now, the original issue was just several coincidences, it appears:

1. `nom` 4 crashed on the config from the readme, making me think it's outdated or some such.
2. I didn't try the config from the readme when testing a build from 2017.
3. Fractional scaling wasn't supported, so the block with fractional scaling involved was ignored.
4. I was using new `sway-output` notation, e.g. `pos 0 0` instead of `position 0,0`, ignored too.

With that out of the way, the config gets parsed fine now. Please, adjust the parsing code according to [this migration guide](https://github.com/Geal/nom/blob/master/doc/upgrading_to_nom_4.md#dealing-with-incomplete-usage) or downgrade to version 3. Also, a comma needs to be replaced with a space here:

https://github.com/emersion/kanshi/blob/1d3f30124358283287bd77dda46bac9d36e8a74f/src/frontend.rs#L40

Some sort of indication of config blocks getting ignored would help with troubleshooting a lot too. Who knows, maybe that's what's happening in #9.

Either way, thanks for the util! Wish Sway wouldn't crash so much on MST...

**Comment by emersion** [comment]
>I've done some extra testing, and, for some reason, now I'm not able to get kanshi to run without panicking at all

Hmm, I'm not able to reproduce with the current code. I've tried using `CompleteByteSlice` and then it always fails with your error.

I spent a few hours trying to upgrade to nom 4, but I'm just getting weird macro errors. I should never have used nom, it sucks, this is not the first time I can't debug a parser issue. Downgraded to nom 3.

>Also, a comma needs to be replaced with a space here

Oh, right, thanks! I don't think I have those setup in my config. Now fixed.
